test/Microsoft.Azure.WebJobs.Host.FunctionalTests/Configuration/WebJobsExtensionOptionsConfigurationTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Azure.WebJobs.Host.TestCommon;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Host.FunctionalTests.Configuration
+{
+    public class WebJobsExtensionOptionsConfigurationTests
+    {
+        [Fact]
+        public void BindOptions_BindsToConfiguration()
+        {
+            var host = new HostBuilder()
+                .ConfigureAppConfiguration(b =>
+                {
+                    b.AddInMemoryCollection(new Dictionary<string, string>
+                    {
+                        { "AzureWebJobs:extensions:test:config1", "test1" },
+                        { "AzureWebJobs:extensions:test:config2", "test2" }
+                    });
+                })
+                .ConfigureDefaultTestHost(b =>
+                {
+                    b.AddExtension<TestExtensionConfigProvider>()
+                    .BindOptions<TestOptions>();
+                }).Build();
+
+            var options = host.Services.GetService<IOptions<TestOptions>>();
+
+            Assert.Equal("test1", options.Value.Config1);
+            Assert.Equal("test2", options.Value.Config2);
+        }
+
+
+        [Fact]
+        public void ConfigureOptions_BindsToConfiguration()
+        {
+            var host = new HostBuilder()
+                .ConfigureAppConfiguration(b =>
+                {
+                    b.AddInMemoryCollection(new Dictionary<string, string>
+                    {
+                        { "AzureWebJobs:extensions:test:config1", "test1" },
+                        { "AzureWebJobs:extensions:test:config2", "test2" }
+                    });
+                })
+                .ConfigureDefaultTestHost(b =>
+                {
+                    b.AddExtension<TestExtensionConfigProvider>()
+                    .ConfigureOptions<TestOptions>((section, o) =>
+                    {
+                        section.Bind(o);
+                        o.Config3 = "called";
+                    });
+                }).Build();
+
+            var options = host.Services.GetService<IOptions<TestOptions>>();
+
+            Assert.Equal("test1", options.Value.Config1);
+            Assert.Equal("test2", options.Value.Config2);
+            Assert.Equal("called", options.Value.Config3);
+        }
```

Comment:
Can we have a test that calls `BindOptions<>` and then `ConfigureOptions<>` in one chain?